internal/app/transport/streamv3.go

```diff
@@ -10,20 +13,28 @@ var _ Stream = &streamv3{}
 type streamv3 struct {
 	grpcClientStream grpc.ClientStream
 	initialRequest   Request
+	logger           log.Logger
 }
 
 // NewStreamV3 creates a new wrapped transport stream
-func NewStreamV3(clientStream grpc.ClientStream, req Request) Stream {
+func NewStreamV3(clientStream grpc.ClientStream, req Request, l log.Logger) Stream {
 	return &streamv3{
 		grpcClientStream: clientStream,
 		initialRequest:   req,
+		logger:           l.Named("stream"),
 	}
 }
 
 func (s *streamv3) SendMsg(version string, nonce string) error {
 	msg := s.initialRequest.GetRaw().V3
 	msg.VersionInfo = version
 	msg.ResponseNonce = nonce
+	s.logger.With(
+		"request_type", msg.GetTypeUrl(),
+		"request_version", msg.GetVersionInfo(),
+		"request_msg", msg.ResourceNames,
```

Comment:
nit: should this be `request_resource_names` since the message is more than the resource names?

Reply:
makes sense. I dont know why i named it request_msg previously.